main.py

```diff
@@ -12,9 +13,13 @@
 FONT = cv2.FONT_HERSHEY_SIMPLEX
 FONTSCALE = 1
 TEXTCOLOUR = (255, 0, 0)
-
-with open(SAVE_FILE, "rb") as f:
-    data = pickle.load(f)
+TOLERANCE = 0.4
+if not os.path.exists(SAVE_FILE):
+    data = {"encodings":[],"names":[]}
+    print("No encoding file exists running face detection only")
+else:
+    with open(SAVE_FILE, "rb") as f:
+        data = pickle.load(f)
 Knownencodings = data["encodings"]
```

Comment:
Knownencodings -> known_encodings

underscores between words with all lower case is called **snake case** and is best for normal variables.